UofS-Pulse-Binfo/uofspb_germplasm
# Adds germplasm collection panel

This PR will add germplasm collection panel (list exportable to a csv file) to study/experiment content.

## Metadata

 - Issue #




Documentation:

## Description


1. Modified local__project_germcollection field to
a. Fetch germplasm/stock collection of a study
b. Create table markup showing germplasm list based on the set collection in the field settings panel.
c. Update install file to implement a hook schema and setup backend used to map collection to a study.
 
## Dependencies



## Testing?




1. Clone module and switch branch to project-collection-field.
2. Install and enable module - germcollection under Tripal Germplasm
3. Modify Study Tripal Content Type and add fields germplasm collection (local: project germcollection)
![Screen Shot 2022-03-02 at 12 16 28 PM](https://user-images.githubusercontent.com/15472253/156423323-8a23bc2d-19dc-4d47-bf7f-cc1ce9e42b90.png)
Click check for new field link if fields have not been added previously.
4. Click save. Note: before saving make sure fields are not in the disabled list in managed display page of the field and set the FORMAT to either inline or above and not hidden (shown below).
![Screen Shot 2022-03-03 at 10 02 06 AM](https://user-images.githubusercontent.com/15472253/156604251-ce3a7ee3-63ed-4186-acfb-3ec70efdf5a0.png)

6. Modify an existing study and set the value under Germplasm field in the update study page.
![Screen Shot 2022-03-02 at 12 19 41 PM](https://user-images.githubusercontent.com/15472253/156423850-3979fab2-258b-477b-9d29-694c8ade0546.png)
7. Click save and reload study page to view germplasm collection list.
![Screen Shot 2022-03-02 at 12 20 57 PM](https://user-images.githubusercontent.com/15472253/156424271-177e4d0d-1430-46c6-bb1d-a9bdcb2f7c10.png)

**Additional configuration:**
1. To view and export germplasm collection, user must be logged in and given permission to view and download germplasm collection.
